### PR TITLE
TandemFoil: T_max=10 — lr=1e-3 (higher LR with rapid cycling)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The current best configuration uses `--lr 3e-4` with `--cosine_t_max 10` (rapid cycling). With fast cosine annealing, the optimizer restarts frequently and the learning rate spends significant time at the peak. A higher peak LR of 1e-3 (3.3x the current baseline) may allow the model to explore a wider loss landscape per cycle, potentially escaping local minima more effectively and finding better solutions within the epoch budget. Lion optimizer's sign-based updates may be particularly compatible with higher learning rates compared to Adam-family optimizers.

## Instructions

Run a single experiment with `--lr 1e-3` and `--cosine_t_max 10`, keeping everything else identical to the golden config. Use `--wandb_group tandem-tmax10-lr-high`.

```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 1e-3 \
  --cosine_t_max 10 \
  --no-use-ema \
  --model_slices 64 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --wandb_group tandem-tmax10-lr-high \
  --epochs 999
```

If this run diverges (loss spikes and doesn't recover), try `--lr 5e-4` as a fallback:
```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 5e-4 \
  --cosine_t_max 10 \
  --no-use-ema \
  --model_slices 64 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --wandb_group tandem-tmax10-lr-high \
  --epochs 999
```

Report the best `val_primary/surface_pressure_mae` for each run attempted and include W&B run IDs.

## Baseline

Current best TandemFoil results (PR #2490, W&B run `77yoba65`):
- `val_primary/surface_pressure_mae`: **75.59**
- Test MAE: **72.12**
- Per-split test MAE: single_in_dist=72.33, geom_camber_rc=76.01, geom_camber_cruise=70.80, re_rand=69.34

Reproduce baseline:
```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 3e-4 \
  --cosine_t_max 10 \
  --no-use-ema \
  --model_slices 64 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --epochs 999
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT kaneda:

## Results

### Summary

| Config | Epochs | Best val_primary | Test primary | W&B Run | Status |
|--------|--------|-----------------|--------------|---------|--------|
| **lr=1e-3** | 9 | 127.16 (epoch 8) | 353.61 (post-divergence) | `chutrgmm` | Diverged E9 |
| **lr=5e-4** (fallback) | 7 | **97.69** (epoch 7) | **92.00** | `hcx7882f` | Stable |
| Baseline (lr=3e-4) | — | **75.59** | **72.12** | `77yoba65` | — |

**Neither run beats the baseline.** lr=3e-4 remains the optimal learning rate for T_max=10 with Lion.

### Epoch-by-epoch val_primary/surface_pressure_mae

| Epoch | lr=1e-3 | lr=5e-4 |
|-------|---------|---------|
| 1 | 234.68 | 200.99 |
| 2 | 193.77 | 156.00 |
| 3 | 169.78 | 128.22 |
| 4 | 213.57 | 112.30 |
| 5 | 150.81 | 111.03 |
| 6 | 135.74 | 105.05 |
| 7 | 162.44 | **97.69** |
| 8 | **127.16** | — |
| 9 | 377.31 (diverged) | — |

### Per-split test breakdown (surface_pressure_mae)

| Split | lr=1e-3 | lr=5e-4 | Baseline (lr=3e-4) |
|-------|---------|---------|---------------------|
| **Primary (eq4)** | 353.61 | 92.00 | **72.12** |
| single_in_dist | 448.43 | 88.66 | **72.33** |
| geom_camber_rc | 407.82 | 96.42 | **76.01** |
| geom_camber_cruise | 243.00 | 88.13 | **70.80** |
| re_rand | 315.17 | 94.79 | **69.34** |

### Commands used

```bash
# lr=1e-3 (diverged)
CUDA_VISIBLE_DEVICES=0 python train.py --dataset tandemfoil --optimizer lion --lr 1e-3 --cosine-t-max 10 --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999 --agent kaneda --wandb-name "kaneda/tmax10-lr1e3" --wandb-group "tandem-tmax10-lr-high"

# lr=5e-4 (fallback)
CUDA_VISIBLE_DEVICES=0 python train.py --dataset tandemfoil --optimizer lion --lr 5e-4 --cosine-t-max 10 --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999 --agent kaneda --wandb-name "kaneda/tmax10-lr5e4-fallback" --wandb-group "tandem-tmax10-lr-high"
```

### What happened

1. **lr=1e-3 diverged at epoch 9.** The run was progressing (best val=127.16 at E8) but showed extreme oscillation throughout — peaks at lr=1e-3 caused val regressions of 30-60 points. At epoch 9, the model catastrophically diverged (val spiked to 377, loss doubled to 0.72). The divergence happened at a trough (lr=0), meaning the damage was inflicted during the prior high-lr phase and only became visible at evaluation. Test metrics are meaningless (353.61) since they were taken post-divergence.

2. **lr=5e-4 was stable but 29% worse than baseline on val.** Smooth monotonic descent over 7 epochs (val: 200→97.69), with negligible oscillation between peaks and troughs. However, 7 epochs at lr=5e-4 only reached 97.69 vs the baseline's 75.59 — suggesting that lr=5e-4 converges slower and would need significantly more epochs to match lr=3e-4. The comparison is confounded by epoch count: the baseline had ~14 epochs while this run only got 7.

3. **Lion optimizer does NOT benefit from higher LRs at T_max=10.** Despite the hypothesis that Lion's sign-based updates would be compatible with higher LRs, the opposite is true: lr=1e-3 (3.3x baseline) is unstable and lr=5e-4 (1.67x baseline) converges slower per-epoch than lr=3e-4. The rapid cosine cycling (T_max=10 = 75 full cycles per epoch) amplifies the effect of lr magnitude — at 1e-3, each cycle pushes the model too far, and the cumulative damage eventually causes divergence.

### Suggested follow-ups
- **lr=4e-4 as a tighter bracket**: If the goal is to find the optimal LR for T_max=10, test 4e-4 as a midpoint between the baseline 3e-4 and the too-high 5e-4.
- **lr=5e-4 with longer T_max**: lr=5e-4 might work better with T_max=30-120 where the lr changes more gradually, reducing the per-cycle disruption.
- **The baseline lr=3e-4 appears optimal for rapid-cycling (T_max=10).** Higher LRs are counterproductive with fast cosine restarts.

W&B group: `tandem-tmax10-lr-high`

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2583 because ADVISOR: Closing — lr=1e-3 diverges at epoch 9 (val→377). Fallback lr=5e-4 only gets 97.69 at 7 epochs. lr=3e-4 is the confirmed optimum for TandemFoil Lion+T_max=10. DEAD END..
